virtio-devices/src/block.rs

```diff
@@ -232,13 +258,43 @@ impl BlockEpollHandler {
                 .ok_or(Error::MissingEntryRequestList)?;
             request.complete_async().map_err(Error::RequestCompleting)?;
 
+            let latency = request.start.elapsed().as_micros() as u64;
+            let read_ops_lst = self.counters.read_ops.load(Ordering::Relaxed) as i64;
+            let write_ops_lst = self.counters.write_ops.load(Ordering::Relaxed) as i64;
+            let mut read_avg =
+                self.counters.read_latency_avg.load(Ordering::Relaxed) as i64 * LATENCY_SCALE;
+            let mut write_avg =
+                self.counters.write_latency_avg.load(Ordering::Relaxed) as i64 * LATENCY_SCALE;
             let (status, len) = if result >= 0 {
                 match request.request_type {
                     RequestType::In => {
                         for (_, data_len) in &request.data_descriptors {
                             read_bytes += Wrapping(*data_len as u64);
                         }
                         read_ops += Wrapping(1);
+                        match latency {
+                            d if d < self.counters.read_latency_min.load(Ordering::Relaxed) => {
+                                self.counters
+                                    .read_latency_min
+                                    .store(latency, Ordering::Relaxed);
+
+                                // Set initial read latency max.
+                                if self.counters.read_latency_max.load(Ordering::Relaxed) == 0 {
+                                    self.counters
+                                        .read_latency_max
+                                        .store(latency, Ordering::Relaxed);
+                                }
+                            }
+                            d if d > self.counters.read_latency_max.load(Ordering::Relaxed) => {
+                                self.counters
+                                    .read_latency_max
+                                    .store(latency, Ordering::Relaxed);
+                            }
+                            _ => {}
+                        }
+                        read_avg = read_avg
+                            + (latency as i64 * LATENCY_SCALE - read_avg)
```

Comment:
Is the scaling constant used to avoid rounding errors?

Reply:
For reducing the precision loss caused by calculate, maybe I should save scaled value in counter. read_avg too.

Reply:
A comment would be sufficient the the scaling factor is just to ensure precision.

Reply:
> A comment would be sufficient the the scaling factor is just to ensure precision.

Which you added! So this is all good.